### ISSUE TITLE
Straightening fails on Linux

### ISSUE THREAD

**Issue by benjamindeleener** [neuropoly/spinalcordtoolbox#472]
On branch bdl_test_straightening, isct_ANTSLandmarksBSplineTransform outputs a wrong warping field, while it works fine on OSX. The resulting image is terrible (t2 from sct_example_data):

![screen shot 2015-08-24 at 11 02 15](https://cloud.githubusercontent.com/assets/7757117/9443559/a96f2514-4a4f-11e5-99d1-53b9260738fe.png)

**Comment by benjamindeleener** [comment]
It is not a issue of dependences because the two system have the same installation of Python library, conda, ITK/VTK and ANTs.

**Comment by jcohenadad** [comment]
that looks like a wrong pairing of labels (e.g., 1-2-3-4 versus 1-4-3-2).

**Comment by benjamindeleener** [comment]
I've compared and they are good and well paired. Same labels on linux and osx.

**Comment by benjamindeleener** [comment]
It seems like ending vector is quite too big:
![screen shot 2015-08-24 at 12 06 24](https://cloud.githubusercontent.com/assets/7757117/9445147/9800347c-4a58-11e5-8235-03eef36ee1e8.png)

And thie reason is that euler transform is not the same on linux and osx:
OSX:
<img width="710" alt="screen shot 2015-08-24 at 12 11 01" src="https://cloud.githubusercontent.com/assets/7757117/9445260/37e9d092-4a59-11e5-88a3-5b14a67ffc5b.png">

Linux:
![screen shot 2015-08-24 at 12 11 17](https://cloud.githubusercontent.com/assets/7757117/9445276/4c9337ea-4a59-11e5-99b5-c2bbcf55a878.png)

**Comment by benjamindeleener** [comment]
ITK transform on:
OSX:
<img width="710" alt="screen shot 2015-08-24 at 12 16 00" src="https://cloud.githubusercontent.com/assets/7757117/9445402/f0fde47e-4a59-11e5-8c41-1719f194b4ed.png">

Linux:
![screen shot 2015-08-24 at 12 16 32](https://cloud.githubusercontent.com/assets/7757117/9445415/fc309b20-4a59-11e5-9071-f7a3cd866699.png)

**Comment by benjamindeleener** [comment]
Note that ITK optimizers are using vnl optimizers and they are probably different on OSX and Linux...

**Comment by benjamindeleener** [comment]
Made some improvements. Incremental integers are not completely the same between linux and osx compiler:

before:
```C++
// Place points in the point sets
curved->SetPoint(pointId, pC);
curved->SetPointData(pointId, pointId);
straight->SetPoint(pointId, pS);
straight->SetPointData(pointId, pointId++);
```

After:
```C++
// Place points in the point sets
curved->SetPoint(pointId, pC);
curved->SetPointData(pointId, pointId);
straight->SetPoint(pointId, pS);
straight->SetPointData(pointId, pointId);
pointId++;
```

Results on
Linux:
![screen shot 2015-08-24 at 14 09 32](https://cloud.githubusercontent.com/assets/7757117/9448347/21e25a60-4a6a-11e5-897d-ff3cc2ec6b94.png)

OSX:
![screen shot 2015-08-24 at 14 09 23](https://cloud.githubusercontent.com/assets/7757117/9448353/26020dca-4a6a-11e5-81d0-0dd236c81acf.png)

There is still a difference, but at least straightening is working on linux.

**Comment by benjamindeleener** [comment]
There is actually no differences between OSX and Linux now:
Linux:
![screen shot 2015-08-24 at 14 17 52](https://cloud.githubusercontent.com/assets/7757117/9448487/02124b2c-4a6b-11e5-9440-626c7b052da2.png)

OSX:
![screen shot 2015-08-24 at 14 17 26](https://cloud.githubusercontent.com/assets/7757117/9448497/0f534142-4a6b-11e5-95ea-1531836043db.png)

The issue is fixed.

**Comment by jcohenadad** [comment]
BRAVO :+1:

**Comment by benjamindeleener** [comment]
Fixed in commit c0f399ab41c802832216984d83a5a2af799d9602
